### ISSUE TITLE
(Multi)FileSelector cannot set check_on_set=True with non-empty/null default

### ISSUE THREAD

**Issue by sdrobert** [holoviz/param#307]
Hi,

Thanks again for the great package. As it stands, Selector initialization happens before the call to `update()`. ``FileSelector`` and ``MultiFileSelector`` can't actually set any non-empty defaults when ``check_on_set=True`` because ``objects`` has yet to be populated. To reproduce:

``` sh
echo "
import param

class Foo(param.Parameterized):
  a = param.FileSelector('foo.py', path='*', check_on_set=True)

" > foo.py
python foo.py
```

Sean

**Comment by tonyfast** [comment]
I can reproduce problem and i receive the following error message.

```
~/miniconda3/lib/python3.7/site-packages/param/__init__.py in _validate(self, val)
   1240             items = '[' + ', '.join(items) + limiter
   1241             raise ValueError("%s not in Parameter %s's list of possible objects, "
-> 1242                              "valid options include %s"%(val,attrib_name, items))
   1243 
   1244     def _ensure_value_is_in_objects(self,val):

ValueError: foo.py not in Parameter None's list of possible objects, valid options include []
```

**Comment by jbednar** [comment]
Fixed in #497
